ceph/rados/core_workflows.py

```diff
@@ -1112,9 +1112,40 @@ def delete_pool(self, pool: str) -> bool:
             log.error(f"Pool:{pool} does not exist on cluster, cannot delete")
             return True
 
+        pool_details = self.run_ceph_command(
+            cmd="ceph osd pool ls detail", client_exec=True
+        )
         cmd = f"ceph osd pool delete {pool} {pool} --yes-i-really-really-mean-it"
         self.client.exec_command(cmd=cmd, sudo=True)
 
+        time.sleep(2)
+        # Improving cleanup for EC pools
+        for pool in pool_details:
+            if pool["pool_name"] == pool:
```

Comment:
We already have a method called `def get_pool_details(self, pool) -> dict:` in core workflows which would return the pool ls detail for the input pool name, we can use this an eliminate the additional traversal using for loop

Reply:
done